python-sdk/tests/sql/operators/test_run_raw_sql.py

```diff
@@ -16,3 +20,154 @@ def test_make_row_serializable(rows):
+
+        @aql.run_raw_sql(results_format="list", conn_id="sqlite_default")
+        def dummy_method():
+            return "SELECT 1+1"
+
+        dummy_method()
+
+    test_utils.run_dag(sample_dag)
+    results_as_list.assert_called_with([])
+
+
+@mock.patch("astro.sql.operators.raw_sql.RawSQLOperator.results_as_pandas_dataframe")
+@mock.patch("astro.databases.base.BaseDatabase.run_sql")
+def test_run_sql_calls_pandas_dataframe_handler(run_sql, results_as_pandas_dataframe, sample_dag):
+    results_as_pandas_dataframe.return_value = []
+    run_sql.return_value = []
+    with sample_dag:
+
+        @aql.run_raw_sql(results_format="pandas_dataframe", conn_id="sqlite_default")
+        def dummy_method():
+            return "SELECT 1+1"
+
+        dummy_method()
+
+    test_utils.run_dag(sample_dag)
+    results_as_pandas_dataframe.assert_called_with([])
+
+
+@mock.patch("astro.sql.operators.raw_sql.RawSQLOperator.results_as_pandas_dataframe")
+@mock.patch("astro.databases.base.BaseDatabase.run_sql")
+def test_run_sql_gives_priority_to_pandas_dataframe_handler(run_sql, results_as_pandas_dataframe, sample_dag):
+    """
+    Test that run_sql calls `results_format` specified handler over handler passed in decorator.
+    """
+    results_as_pandas_dataframe.return_value = []
+    run_sql.return_value = []
+    with sample_dag:
+
+        @aql.run_raw_sql(
+            results_format="pandas_dataframe", conn_id="sqlite_default", handler=lambda x: x.fetchall()
+        )
+        def dummy_method():
+            return "SELECT 1+1"
+
+        dummy_method()
+
+    test_utils.run_dag(sample_dag)
+    results_as_pandas_dataframe.assert_called_with([])
+
+
+@mock.patch("astro.sql.operators.raw_sql.RawSQLOperator.results_as_pandas_dataframe")
+@mock.patch("astro.databases.base.BaseDatabase.run_sql")
+def test_run_sql_called_handler(run_sql, results_as_pandas_dataframe, sample_dag):
+    """
+    Test that run_sql calls `handler` passed in decorator.
+    """
+    results_as_pandas_dataframe.return_value = []
+    return_value = [1, 2, 3]
+    run_sql.return_value = return_value
+
+    def verify(result):
+        assert result == return_value
+        return result
+
+    with sample_dag:
+
+        @aql.run_raw_sql(conn_id="sqlite_default", handler=verify)
+        def dummy_method():
+            return "SELECT 1+1"
+
+        dummy_method()
+    test_utils.run_dag(sample_dag)
+
+
+@mock.patch("astro.sql.operators.raw_sql.RawSQLOperator.results_as_pandas_dataframe")
+@mock.patch("astro.databases.base.BaseDatabase.run_sql")
+def test_run_sql_should_raise_exception(run_sql, results_as_pandas_dataframe, sample_dag):
+    """
+    Test that run_sql should raise an exception when fail_on_empty=False
+    """
+    results_as_pandas_dataframe.return_value = []
+    return_value = [1, 2, 3]
+    run_sql.return_value = return_value
+
+    def raise_exception(result):
+        raise ValueError("dummy exception")
+
+    with pytest.raises(ValueError) as e:
```

Comment:
```suggestion
    with pytest.raises(ValueError, match="dummy exception"):
```

https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

Reply:
Added in a separate comment since there was one more line which we needed to remove.